### ISSUE TITLE
Helpers don't update if there are notFound routes (flow-router)

### ISSUE THREAD

**Issue by serkandurusoy** [zimme/meteor-active-route#39]
I'm using flow-router (with blaze-layout) and have a couple of routes.

I'm using the `isActiveRoute` helper within a dynamically rendered navigation template.

Given

* I browse using the links in the nav
* one of the links is broken
* I enter a route (with helper to mark the nav link as active)
* I click on the broken link (see the not found page, which uses the same navigation template)

I expect that the previous helper return `false`, but it continues returning `active`

In the meantime, `ActiveRoute.name()` for the same route return `false` as it should

At this point, it takes a few back and forth clicking between other nav links for the helper to pick up.

I don't know if this is a flow-router issue, or with the helper. I'll also cross-post it over at the flow-router repo.

**Comment by zimme** [comment]
It would be really helpful to get a reproduction repo for this.

**Comment by serkandurusoy** [comment]
@zimme this may actually be flow-router related after all. Arunoda is looking into this over at the issue I've cross posted.

Do you still think there is something you could do?

**Comment by zimme** [comment]
Yes, I just saw the conversation on the related issue. If it's a special route, maybe it's something I can look for in my code.

I'm doing a re-write for `3.0.0` (es6) and once it's released I'll get back to this and hopefully the notFound route is working as a regular route by then otherwise I'll see if there's something I can do on my end to work around the problem.

I'll open a PR brach for `3.0.0` and if you figure out a way to work around this problem while I'm working on that feel free to submit a PR against that branch.

**Comment by serkandurusoy** [comment]
I don't know if you can detect the notFound route being hit, but what I realize is, even though the `{{isActiveRoute 'routeName'}}` somehow `ActiveRoute.name('routeName')` works and `FlowRouter.getRouteName()`returns undefined in the meantime.

Since flow router's notFound route is not a proper 404, it does not provide much value, therefore, I've decided to work around the problem by defining a catch all route at the very end of the route definitions:

```javascript
FlowRouter.route('/:notFound*', {
  name: 'notFound',
  action: FlowRouter.notFound.action()
});
```

But I guess this does not present any PR value. But I'll keep a close eye on this and get back to you if I can manage to get to the bottom of it in a more proper way.

Thanks.

**Comment by zimme** [comment]
If `ActiveRoute.name` works but `{{isActiveRoute ...}}` doesn't reactively change when `notFound` route are hit then flow-router most likely doesn't invalidate when hitting `notFound` routes.

**Comment by zimme** [comment]
So putting `ActiveRoute.name` inside a `Tracker.autorun` gives me the same kind of "problem". I think this is related to how flow-router invalidates path changes when hitting the notFound route.

**Comment by serkandurusoy** [comment]
Perhaps it is related to Arunoda's comment about invalidating `watchPathChange()`

**Comment by davidlawson** [comment]
@serkandurusoy I think you mean `action: FlowRouter.notFound.action` (no parentheses). Still only a temporary fix, of course!

**Comment by zimme** [comment]
So has this been resolved in flow-router yet? as I believe we came to the conclusion that this was a flow-router problem, no?
